tracetools/CMakeLists.txt

```diff
@@ -81,6 +81,10 @@ if(TRACETOOLS_LTTNG_ENABLED)
     target_link_libraries(${PROJECT_NAME} "-rdynamic")
   endif()
 endif()
+if(NOT WIN32)
```

Comment:
Haven't been able to get Windows CI to run yet, but I think it will fail. I think this should be:

```suggestion
if(NOT TRACETOOLS_DISABLED AND NOT WIN32)
```

If `TRACETOOLS_DISABLED`, `tracetools` is only an interface library, and so `target_link_libraries()` must be called with `INTERFACE`. We don't really need to link against `dl` in that case anyway, so we should just avoid doing this if `TRACETOOLS_DISABLED`. @cottsay thoughts?

Reply:
> Haven't been able to get Windows CI to run yet, but I think it will fail.

If it passed before, why would it fail with this PR? This block is only processed on non-Windows platforms.

> If `TRACETOOLS_DISABLED`, tracetools is only an interface library, and so `target_link_libraries()` must be called with `INTERFACE`.

I think you're right. We can test it with a quick Linux CI build though: [![Build Status](https://ci.ros2.org/buildStatus/icon?job=ci_linux&build=18141)](https://ci.ros2.org/job/ci_linux/18141/)

Reply:
> If it passed before, why would it fail with this PR? This block is only processed on non-Windows platforms.

Yeah, I quickly realized and edited my comment :sweat_smile:

Reply:
Alright, I changed it to `if(NOT TRACETOOLS_DISABLED)`. I'll trigger CI again.

Reply:
> We can test it with a quick Linux CI build though: [![Build Status](https://camo.githubusercontent.com/0da22f70eb94ccadddc02702c88e4676001f5790aa18524cea6eaae0e62c57db/68747470733a2f2f63692e726f73322e6f72672f6275696c645374617475732f69636f6e3f6a6f623d63695f6c696e7578266275696c643d3138313431)](https://ci.ros2.org/job/ci_linux/18141/)

This should've been `-DTRACETOOLS_DISABLED:BOOL=ON` instead of `-DTRACETOOLS_DISABLED:BOOL=OFF` to test this, but yeah we don't really need to test this.

Reply:
:facepalm: I'll cancel it.